staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -309,15 +333,33 @@ const (
 // WaitForNamedCacheSync is a wrapper around WaitForCacheSync that generates log messages
 // indicating that the caller identified by name is waiting for syncs, followed by
 // either a successful or failed sync.
+//
+// TODO: logcheck:context // WaitForNamedCacheSyncWithContext should be used instead of WaitForNamedCacheSync in code which supports contextual logging.
```

Comment:
This comment irritates me. What is the downside to switch to the context variant everywhere? This function is a little special because it has the controllerName argument. Is it the only thing we lose when switching in legacy code?

Also, what is the deprecation pattern? Shouldn't we consequently deprecate non-context variants?

Reply:
There is no downside, but it's not required. We don't intend to remove the traditional non-contextual variants and therefore they are not deprecated.

If we were to mark them as deprecated even though there is no need to stop using them, then we  would cause a lot of unnecessary work downstream because well-meaning contributors will start creating PRs which then need to be reviewed by maintainers, or maintainers themselves feel compelled to update their code.

This approach with `logcheck:context` avoids that by only triggering warnings in those projects that care. It's using a staggered rollout (first add new API, start using it in Kubernetes, then remove the TODO) to keep the PRs within a manageable size - see https://github.com/kubernetes/kubernetes/issues/126379.

Reply:
@soltysh: ^^^ this explains why I will copy you on some future follow-up PRs.

Reply:
I really don't like the approach of putting TODOs in interface godoc that will show up for users

Reply:
My hope was that we could get this PR merged, users in Kubernetes updated, and then remove the TODO all before 1.32 goes out. https://github.com/kubernetes/kubernetes/issues/126379 explains the process.

The alternative is to update all users directly in this PR, which would make it quite a bit larger.

Would the comment be okay if it left out the TODO,? For example:
```
// Contextual logging: WaitForNamedCacheSyncWithContext should be used instead of WaitForNamedCacheSync in code which supports contextual logging.
```

Later that would get replaced with `// logcheck:context // WaitForNamedCacheSyncWithContext should be used ...`.

Reply:
I just don't want TODOs intended for us in user-facing godoc ... whether that means omitting them, or putting them in a different spot, I don't really care

Reply:
So you are okay with `// Contextual logging: WaitForNamedCacheSyncWithContext ....`?

That is user-facing information which belongs into the API docs.

Reply:
Changed (https://github.com/kubernetes/kubernetes/compare/f2dcad6300494ce13ee9e9a0d60b090c19aeaf52..42571dbc3e67ab16ac6ab2256a653b00d2679a8a).